iwyu_ast_util.cc

```diff
@@ -433,6 +438,14 @@ const DeclContext* GetDeclContext(const ASTNode* ast_node) {
   return nullptr;
 }
 
+const CallExpr* GetCallExpr(const ASTNode* node) {
```

Comment:
The function name is a little too generic here, since we can't lean on the type system to know what kind of ASTNode we can pass in. Maybe `GetConstructCallExpr` to narrow it down a little?

Reply:
`GetConstructCallExpr` name may erroneously suggest that it returns a constructor call expression, but it actually returns function call expression which involves its argument construction. I hope that this function may be used for getting `CallExpr` for any direct argument node, not only `CXXConstructExpr`. However, what is "direct" is questionable, in the presence of `CXXBindTemporaryExpr` special handling...

Reply:
Maybe, `GetCallExprOfArgument`?

Reply:
But, yeah, `CXXBindTemporaryExpr` handling is present here just because this function handles argument construction. So, `GetCallExprForConstructedArgument` or something shorter?

Reply:
Or, maybe, it should be just a lambda inside `SourceOrTargetTypeIsProvided`?

Reply:
Yeah, it's probably better to inline it there if it's hard to name it.

`SourceOrTargetTypeIsProvided` is also kind-of dangerously named, but it's a difficult enough name that you're not tempted to call it on any old `ASTNode*` :-)